geode-core/src/test/java/org/apache/geode/internal/cache/DistributedRegionTest.java

```diff
@@ -180,4 +189,75 @@ public void regionSyncNotInvokedInPerformSynchronizeForLostMemberTaskIfRegionNot
 
     verify(distributedRegion, never()).synchronizeForLostMember(member, lostMemberVersionID);
   }
+
+  @Test
+  public void validateAsynchronousEventDispatcherShouldDoNothingWhenDispatcherIdCanNotBeFound() {
+    InternalCache internalCache = mock(InternalCache.class);
+    when(internalCache.getAllGatewaySenders())
+        .thenReturn(Collections.singleton(mock(GatewaySender.class)));
+    DistributedRegion distributedRegion = mock(DistributedRegion.class);
```

Comment:
It might be possible but it will require some effort as the `DistributedRegion` class doesn't have a default constructor and the initialisation logic is not trivial (I tried to do it but need to mock a lot of other objects to avoid getting all kind of exceptions)... I can do it if it becomes a blocker to merge this `PR`, otherwise I prefer to use the same approach the rest of this test class is using.

Reply:
Having looked at the constructor for `DistributedRegion` I can see that it would be way, way too much work to create a spy. The current approach is fine.

Reply:
👍